mlflow/utils/search_utils.py

```diff
@@ -303,9 +306,10 @@ def _get_value_for_sort(cls, run, key_type, key, ascending):
                                   error_code=INVALID_PARAMETER_VALUE)
 
         # Return a key such that None values are always at the end.
+        is_null_or_nan = sort_value is None or math.isnan(sort_value)
```

Comment:
Looks like tests caught an important case: `sort_value` may correspond to a param, tag, or attribute key. In these cases, we use a lexicographic sort. 

Failed test job: https://travis-ci.org/mlflow/mlflow/jobs/559532493
Stacktrace:
```
tests/utils/test_search_utils.py:279: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mlflow/utils/search_utils.py:329: in sort
    reverse=not ascending)
mlflow/utils/search_utils.py:328: in <lambda>
    key=lambda run: cls._get_value_for_sort(run, key_type, key, ascending),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cls = <class 'mlflow.utils.search_utils.SearchUtils'>
run = <Run: data=<RunData: metrics={'key1': 125}, params={'my_param': 'B'}, tags={'t...', run_id='7', run_uuid='7', start_time=1, status='FAILED', user_id='user-id'>>
key_type = 'parameter', key = 'my_param', ascending = True
    @classmethod
    def _get_value_for_sort(cls, run, key_type, key, ascending):
        """Returns a tuple suitable to be used as a sort key for runs."""
        sort_value = None
        if key_type == cls._METRIC_IDENTIFIER:
            sort_value = run.data.metrics.get(key)
        elif key_type == cls._PARAM_IDENTIFIER:
            sort_value = run.data.params.get(key)
        elif key_type == cls._TAG_IDENTIFIER:
            sort_value = run.data.tags.get(key)
        elif key_type == cls._ATTRIBUTE_IDENTIFIER:
            sort_value = getattr(run.info, key)
        else:
            raise MlflowException("Invalid order_by entity type '%s'" % key_type,
                                  error_code=INVALID_PARAMETER_VALUE)
    
        # Return a key such that None values are always at the end.
>       is_null_or_nan = sort_value is None or math.isnan(sort_value)
E       TypeError: a float is required
```